[api-minor] Remove all the useless telemetry stuff in the viewer (bug 1802468)

calixteman opened the pull request:
Add a deprecation notification for PDFDocumentLoadingTask.onUnsupportedFeature and PDFDocumentProxy.stats
which are likely useless.
The unsupported feature stuff have initially been added in (#4048) in order to be able to display a
warning bar and to help to have some numbers to know how a feature was used.
Those data are no more used in Firefox.

Snuffleupagus commented:
The main question, which is obviously extremely difficult to answer, is whether there's any third-party users that depend on the API-functionality that you're removing here?
Most likely the `stats`-getter is basically unused, however the `onUnsupportedFeature`-callback may be used (by those that want to do custom error-handling).

At the very least this is an `[api-minor]` change, and we probably need to `deprecate` it properly (for a minimum one release) before removing it. Perhaps we should only make the `web/`-folder changes for now?

calixteman committed:
[api-minor] Remove all the useless telemetry stuff in the viewer (bug 1802468)

Add a deprecation notification for PDFDocumentLoadingTask.onUnsupportedFeature and PDFDocumentProxy.stats
which are likely useless.
The unsupported feature stuff have initially been added in (#4048) in order to be able to display a
warning bar and to help to have some numbers to know how a feature was used.
Those data are no more used in Firefox.

Snuffleupagus commented:
/botio integrationtest

Snuffleupagus reviewed (approved):
Please also remove the following, now unused, lines:
 - https://github.com/mozilla/pdf.js/blob/f96b99c30cd3a37edfd7f5f58f4e3049539ae1df/web/app.js#L260
 - https://github.com/mozilla/pdf.js/blob/f96b99c30cd3a37edfd7f5f58f4e3049539ae1df/web/app.js#L869

r=me, with those changes; thank you!

calixteman commented:
/botio integrationtest
